src/Uno.Toolkit.RuntimeTests/Tests/ShadowContainerTests.cs

```diff
@@ -137,6 +139,72 @@ public async Task Outer_Shadows(int offsetX, int offsetY, bool inner)
 			}
 		}
 #endif
+
+		[TestMethod]
+		public async Task _AsdAsdReLayoutsAfterChangeInSize()
+		{
+			ShadowContainer.ClearCache();
+
+			var internalBorder = new Border { Height = 200, Width = 200, Background = new SolidColorBrush(Colors.Green) };
+			bool createdNewCanvas = false;
+
+			var shadowContainer = new ShadowContainer
+			{
+				Content = internalBorder,
+				Shadows = new ShadowCollection
+				{
+					new UI.Shadow
+					{
+						Color = Colors.Blue,
+						OffsetX = 10,
+						OffsetY = 10,
+						IsInner = false,
+						Opacity = 1,
+
+					}
+				}
+			};
+
+			UnitTestsUIContentHelper.Content = shadowContainer;
+
+			shadowContainer.SurfacePaintCompleted += SurfacePaintCompleted; ;
+
+			shadowContainer.Shadows.Add(new UI.Shadow
+			{
+				Color = Colors.Red,
+				OffsetX = 10,
+				OffsetY = 10,
+				IsInner = false,
+				Opacity = 1,
+			});
+
+			await UnitTestsUIContentHelper.WaitForIdle();
+			await UnitTestsUIContentHelper.WaitForLoaded(shadowContainer);
+			createdNewCanvas.Should().BeTrue();
+			createdNewCanvas = false;
+
+			// Changing the height of the children content of the shadow. 
+			internalBorder.Height = 400;
+
+			// WaitForIdle doesnt work when called a second time on WinUI.
+			var idleTask = UnitTestsUIContentHelper.WaitForIdle();
+			var delayTask = Task.Delay(1000);
```

Comment:
Not sure if there's a better way to do this.

Reply:
It's working fine on android too, let's see in CI time.

Reply:
The Runtime tests inside of Uno has a [WaitForRelayouted](https://github.com/unoplatform/uno/blob/52fd1995ec5911d56dcea44fe6e48f6bab3c992d/src/Uno.UI.RuntimeTests/IntegrationTests/common/TestServices.WindowHelper.cs#L211), could it be useful here?

Reply:
I believe not because the element had already been loaded previously.
We are waiting for an update on the Height.
But works with the await UnitTestsUIContentHelper.WaitForIdle();

Reply:
WaitForRelayout waits for the LayoutUpdated, not Loaded so this may work for when height changes